From: Arnd Bergmann <arnd@arndb.de>
Subject: kbuild: enable -Wstringop-truncation globally
Date: Mon, 15 Apr 2024 14:20:37 +0200

The remaining warnings of this type have been addressed, so it can now be
enabled by default, rather than only for W=1.

Link: https://lkml.kernel.org/r/20240415122037.1983124-7-arnd@kernel.org
Signed-off-by: Arnd Bergmann <arnd@arndb.de>
Reviewed-by: Kees Cook <keescook@chromium.org>
Cc: Masahiro Yamada <masahiroy@kernel.org>
Cc: Nathan Chancellor <nathan@kernel.org>
Cc: Nicolas Schier <nicolas@fjasle.eu>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 scripts/Makefile.extrawarn |    1 -
 1 file changed, 1 deletion(-)

--- a/scripts/Makefile.extrawarn~kbuild-enable-wstringop-truncation-globally
+++ a/scripts/Makefile.extrawarn
@@ -108,7 +108,6 @@ else
 KBUILD_CFLAGS += $(call cc-disable-warning, format-overflow-non-kprintf)
 KBUILD_CFLAGS += $(call cc-disable-warning, format-truncation-non-kprintf)
 endif
-KBUILD_CFLAGS += $(call cc-disable-warning, stringop-truncation)
 
 KBUILD_CFLAGS += -Wno-override-init # alias for -Wno-initializer-overrides in clang
 
_
